### PR TITLE
RPC: add prepare execution payload

### PR DESCRIPTION
**Pull request by terencechain**
This PR implements get payload for proposer RPC. Outlined in spec:
```python
def prepare_execution_payload(state: BeaconState,
                              pow_chain: Dict[Hash32, PowBlock],
                              finalized_block_hash: Hash32,
                              suggested_fee_recipient: ExecutionAddress,
                              execution_engine: ExecutionEngine) -> Optional[PayloadId]:
    if not is_merge_transition_complete(state):
        is_terminal_block_hash_set = TERMINAL_BLOCK_HASH != Hash32()
        is_activation_epoch_reached = get_current_epoch(state) >= TERMINAL_BLOCK_HASH_ACTIVATION_EPOCH
        if is_terminal_block_hash_set and not is_activation_epoch_reached:
            # Terminal block hash is set but activation epoch is not yet reached, no prepare payload call is needed
            return None

        terminal_pow_block = get_terminal_pow_block(pow_chain)
        if terminal_pow_block is None:
            # Pre-merge, no prepare payload call is needed
            return None
        # Signify merge via producing on top of the terminal PoW block
        parent_hash = terminal_pow_block.block_hash
    else:
        # Post-merge, normal payload
        parent_hash = state.latest_execution_payload_header.block_hash

    # Set the forkchoice head and initiate the payload build process
    payload_attributes = PayloadAttributes(
        timestamp=compute_timestamp_at_slot(state, state.slot),
        prev_randao=get_randao_mix(state, get_current_epoch(state)),
        suggested_fee_recipient=suggested_fee_recipient,
    )
    return execution_engine.notify_forkchoice_updated(parent_hash, finalized_block_hash, payload_attributes)
```

### CODE REVIEW

File: beacon-chain/rpc/prysm/v1alpha1/validator/proposer_execution_payload.go [hunk 1]
```diff
@@ -8,11 +8,100 @@ import (
...
+	if err != nil {
+		return nil, err
+	}
+
+	finalizedBlock, err := vs.BeaconDB.Block(ctx, bytesutil.ToBytes32(st.FinalizedCheckpoint().Root))
+	if err != nil {
+		return nil, err
+	}
+	finalizedBlockHash := params.BeaconConfig().ZeroHash[:]
+	if finalizedBlock != nil && finalizedBlock.Version() == version.Bellatrix {
```
Comment (rauljordan) [inline beacon-chain/rpc/prysm/v1alpha1/validator/proposer_execution_payload.go:77]: This is not safe for future forks, let's make it more lenient

Comment (rauljordan) [inline beacon-chain/rpc/prysm/v1alpha1/validator/proposer_execution_payload.go:77]: If finalized block is nil, is the finalized block supposed to be the genesis block ?

Comment (rauljordan) [reply beacon-chain/rpc/prysm/v1alpha1/validator/proposer_execution_payload.go:77]: Let's add some coverage for this section

Comment (terencechain) [reply beacon-chain/rpc/prysm/v1alpha1/validator/proposer_execution_payload.go:77]: It's not unsafe for a future fork. You could have a Bellatrix block along with an Altair block as finalized. In this case, as specified in the spec, we should use `params.BeaconConfig().ZeroHash`. We definitely do not want to return an error

Comment (terencechain) [reply beacon-chain/rpc/prysm/v1alpha1/validator/proposer_execution_payload.go:77]: This coverage is fully covered. See test case: `transition completed, happy case`. The transition state contains a post-Bellatrix finalized block in DB

Comment (terencechain) [reply beacon-chain/rpc/prysm/v1alpha1/validator/proposer_execution_payload.go:77]: > If finalized block is nil, is the finalized block supposed to be the genesis block ?

I separated out the nil case for a proper error, it's better

**Pull request merged**